components/tools/OmeroJava/test/integration/DuplicationTest.java

```diff
@@ -0,0 +1,1112 @@
+        Assert.assertEquals(duplicate.getTheT().getValue(), original.getTheT().getValue());
+        Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue());
+        Assert.assertEquals(duplicate.getX1().getValue(), original.getX1().getValue());
+        Assert.assertEquals(duplicate.getY1().getValue(), original.getY1().getValue());
+        Assert.assertEquals(duplicate.getX2().getValue(), original.getX2().getValue());
+        Assert.assertEquals(duplicate.getY2().getValue(), original.getY2().getValue());
+    }
+
+    /**
+     * Assert that the given instances have the same property values.
+     * @param original the original instance
+     * @param duplicate the duplicate instance that is expected to correspond to the original
+     */
+    private static void assertSameProperties(Point original, Point duplicate) {
+        Assert.assertEquals(duplicate.getTheZ().getValue(), original.getTheZ().getValue());
+        Assert.assertEquals(duplicate.getTheT().getValue(), original.getTheT().getValue());
+        Assert.assertEquals(duplicate.getTheC().getValue(), original.getTheC().getValue());
+        Assert.assertEquals(duplicate.getCx().getValue(), original.getCx().getValue());
+        Assert.assertEquals(duplicate.getCy().getValue(), original.getCy().getValue());
+    }
+
+    /**
+     * Get the IDs of an image's annotations.
+     * @param imageId the ID of an image
+     * @return the IDs of the annotations on the image
+     * @throws ServerError unexpected
+     */
+    private Set<Long> getImageAnnotations(long imageId) throws ServerError {
+        final Set<Long> annotationIds = new HashSet<Long>();
+        for (final List<RType> result : iQuery.projection(
+                "SELECT child.id FROM ImageAnnotationLink WHERE parent.id = :id",
+                new ParametersI().addId(imageId))) {
+            annotationIds.add(((RLong) result.get(0)).getValue());
+        }
+        return annotationIds;
+    }
+
+    private String followLongAnnotations(LongAnnotation start, int depth) throws ServerError {
+        if (depth < 1) {
+            return "";
+        }
+        final StringBuilder sb = new StringBuilder();
+        sb.append('[');
+        sb.append(start.getLongValue().getValue());
+        for (final IObject result : iQuery.findAllByQuery(
+                "SELECT l.child FROM AnnotationAnnotationLink AS l WHERE l.parent.id = :id ORDER BY l.child.longValue",
+                new ParametersI().addId(start.getId().getValue()))) {
+            if (result instanceof LongAnnotation) {
+                sb.append(followLongAnnotations((LongAnnotation) result, depth - 1));
+            }
+        }
+        sb.append(']');
+        return sb.toString();
+    }
+
+    /**
+     * Create then duplicate an image and test that the properties of the instances in it and its subgraph are correctly duplicated.
+     * @throws Exception unexpected
```

Comment:
Indeed, anything that somehow lives in an original file will wait for 5.3, it shouldn't present much of a problem though.